### ISSUE TITLE
[BUG] deflisten and multiple windows (sub-processes are not killed?)

### ISSUE THREAD

**Issue by Catvert** [elkowar/eww#360]
## Checklist before submitting an issue
- [x] I have searched through the existing [closed and open issues](https://github.com/elkowar/eww/issues?q=is%3Aissue) for eww and made sure this is not a duplicate
- [x] I have specifically verified that this bug is not a common [user error](https://github.com/elkowar/eww/issues?q=is%3Aissue+label%3Ano-actual-bug+is%3Aclosed)
- [x] I am providing as much relevant information as I am able to in this bug report (Minimal config to reproduce the issue for example, if applicable)

## Describe the bug
Hello, I'm trying to create 2 bars (like polybar) for my 2 monitors.
I have some script that poll some informations, like current volume and the playerctl status.

However, when I'm closing the bars (with close-all), some of these sub-processes are leaked out and not closed properly. This 'leak' does not appear with only 1 window (bar) opened. I've also noticed that when I'm opening the 2 window, the processes that are defined in 'deflisten' spawn 2 times instead of one (I'm not sure if this is the expected behavior).

## Reproducing the issue
For example, here is my deflistens :
```
(deflisten music_value :initial "" "eww-music-subscribe")
(deflisten volume_value :initial "0" "eww-volume")
```
eww-volume and eww-music-subscribe are simply scripts that call 'playerctl --follow' and 'pactl subscribe' and pretty-print the output.

Here is my windows : 
```
(defwindow bar_screen1
 :monitor 0
 :geometry (geometry :x 0 :y 0 :width "100%" :height "30px")
 :anchor "top left"

 :windowtype "dock"
 :stacking "bg"
 :wm-ignore false

 :reserve (struts :distance "32px" :side "top")
 (bar :screen 0))


(defwindow bar_screen2
 :monitor 1
 :geometry (geometry :x 0 :y 0 :width "100%" :height "30px")
 :anchor "top left"

 :windowtype "dock"
 :stacking "bg"
 :wm-ignore false

 :reserve (struts :distance "32px" :side "top")
 (bar :screen 1))
```

When opening just one window (bar_screen1) :
![image](https://user-images.githubusercontent.com/8643204/143212670-2bcd3a10-8677-4615-a91f-756c6c11f575.png)

When opening the two windows (bar_screen1 + bar_screen2) : 

![image](https://user-images.githubusercontent.com/8643204/143213009-abba2195-9f9c-44c0-98f4-0daf21904584.png)

Note that eww-volume and eww-music-subscribe (+ tail) are duplicated (Which I think is not an expected behavior, aren't the variables 'global'?)

Now, when I ```eww close-all```, here is the result : 
![image](https://user-images.githubusercontent.com/8643204/143213517-abc348c5-aa35-492c-b7e4-8e8b397298f9.png)

A 'pair' of sub-processes are not killed (which is not normal).

If instead I kill the daemon (eww kill), the subprocesses are still active while eww no longer exists.

![image](https://user-images.githubusercontent.com/8643204/143213880-be32fc86-461c-4025-924b-d4d686359f54.png)

I noticed this bug when after a morning of tweaking my dotfiles, I noticed that I had about 20 processes of 'eww-volume and eww-music-subscribe' in my task manager

I hope my explanations are clear, don't hesitate to come back to me if you need more information.

Thank you for your help.

**Comment by Lisenaaaa** [comment]
bumping this, i use playerctl for spotify song data and it isn't only annoying to have a ton of playerctl processes, it straight up stops it from working if there's too many open

![image](https://user-images.githubusercontent.com/69115669/184430922-29d68e21-8f92-444b-aa9b-03976cf64589.png)

i'd really like it if this were fixed, as i don't really want to have to go through and kill every playerctl process every so often

**Comment by elkowar** [comment]
with a bit of luck this might have been fixed by 4004362633cf21bc00f00b7e261d1a6956ebc844 -- although I haven't been able to really reproduce the issue, so I'm mostly just guessing here ^^'

**Comment by beyond9thousand** [comment]
that doesn't fix it @elkowar
